platform/viewer/cypress/integration/common/OHIFStudyViewer.spec.js

```diff
@@ -15,33 +22,80 @@ describe('OHIF Study Viewer Page', () => {
       cy.get('.ThumbnailEntryContainer:nth-child(2)') //element to be dragged
         .drag('.cornerstone-canvas'); //dropzone element
       
-      const overlaySeriesInformation = 'div.ViewportOverlay > div.bottom-left.overlay-element > div';
-      const expectedText = 'Ser: 2Img: 1 1/13512 x 512Loc: -17.60 mm Thick: 3.00 mm';
-        
-      cy.get(overlaySeriesInformation)
-        .should('have.text', expectedText);
+      const expectedText = 'Ser: 2Img: 1 1/13512 x 512Loc: -17.60 mm Thick: 3.00 mm';  
+      cy.get('@viewportInfoBottomLeft')
+        .should('contain.text', expectedText);
     });
 
     it('checks if Series left panel can be hidden/displayed', ()=> {
-      const seriesButton = '.pull-left > .RoundedButtonGroup > .roundedButtonWrapper > .roundedButton';
-      const leftPanel = 'section.sidepanel.from-left';
+      cy.get('@seriesBtn').click(); 
+      cy.get('@seriesPanel').should('not.be.enabled')
 
-      cy.get(seriesButton).click(); 
-      cy.get(leftPanel).should('not.be.enabled')
-
-      cy.get(seriesButton).click();
-      cy.get(leftPanel).should('be.visible');
+      cy.get('@seriesBtn').click();
+      cy.get('@seriesPanel').should('be.visible');
     });
 
     it('checks if Measurements right panel can be hidden/displayed', ()=> {
-      const measurementsButton = '.pull-right > .RoundedButtonGroup > .roundedButtonWrapper > .roundedButton';
-      const rightPanel = 'section.sidepanel.from-right';
-
-      cy.get(measurementsButton).click();
-      cy.get(rightPanel).should('be.visible');
+      cy.get('@measurementsBtn').click();
+      cy.get('@measurementsPanel').should('be.visible');
       
-      cy.get(measurementsButton).click();
-      cy.get(rightPanel).should('not.be.enabled');
+      cy.get('@measurementsBtn').click();
+      cy.get('@measurementsPanel').should('not.be.enabled');
+
     });
 
+    it('checks if measurement item can be Relabeled under Measurements panel', () => {
+      cy.addLengthMeasurement(); //Adding measurement in the viewport
+      cy.get('@measurementsBtn').click();
+      cy.get('.measurementItem').click();
+      cy.get('.btnAction')
+        .contains('Relabel')
+        .then(($btn) =>{ //click on Relabel button
+          cy.wrap($btn)
+            .click()
+            .then(() =>{
+              cy.get('.searchInput') //search for Bone
+                .should('be.visible')
+                .type('Bone')
+                .then(() =>{
+                  cy.get('.treeInputs > .wrapperLabel') //select 'Bone' result
+                    .contains('Bone')
+                    .click()
+                    .then(() => {
+                      cy.get('.checkIconWrapper').should('be.visible').click(); //confirm the selection on overlay
+                    });
+                });
+            });
+        });
```

Comment:
My biggest hang-up is the nesting here. Does something like this work?

```js
cy.addLengthMeasurement(); //Adding measurement in the viewport
      cy.get('@measurementsBtn').click();
      cy.get('.measurementItem').click();

      // Click "Relabel"
      cy.get('.btnAction')
        .contains('Relabel')
        .click();

      // Search for "Bone"
      cy.get('.searchInput')
        .should('be.visible')
        .type('Bone');

      // Select "Bone" Result
      cy.get('.treeInputs > .wrapperLabel')
         .contains('Bone')
         .click();

       // Confirm Selection
       cy.get('.checkIconWrapper')
          .should('be.visible')
          .click();
```

Alternatives for selecting if you feel locked in to wrap:

- https://github.com/cypress-io/cypress/issues/3745#issuecomment-474211567

Are we using the `should('be.visible')` checks to wait for the animations?

Reply:
Great callout! I replaced the nesting for your suggestion, they both work fine.
You have nice point for the `should('be.visible')` checks. I was using it to make sure that we are clicking on the right-visible element and not on something behind in case it was not visible, but I just realized that the element MUST be visible in order to be retrieved by `cy.get`, therefore, `should('be.visible')` is redundant here. We can remove them. :)